google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2761,4 +2790,77 @@ private void initMetricsCollection(MetricRegistry metricRegistry, List<LabelValu
         // TODO: Remove metric.
         ignored -> 0L);
   }
+
+  private void initOpenTelemetryMetricsCollection(
+      OpenTelemetry openTelemetry, Attributes attributes) {
+
+    if (openTelemetry != null) {
+      Meter meter = openTelemetry.getMeter(MetricRegistryConstants.Scope);
+
+      meter
+          .gaugeBuilder(MAX_ALLOWED_SESSIONS_NAME)
+          .setDescription(MAX_ALLOWED_SESSIONS_DESCRIPTION)
+          .setUnit(COUNT)
+          .buildWithCallback(
+              measurement -> {
+                measurement.record(options.getMaxSessions(), attributes);
```

Comment:
When is this callback being called? Because `options.getMaxSessions()` will never change during the lifetime of the session pool, so if there is something like `buildWithConstant(...)` or similar, then that would probably be a better fit here.

Reply:
I double checked this with OpenTelemetry team. What we are doing here is the correct way of handling such data, even if its constant. 
Even if its a constant value, a callback is required every few seconds to get the current value, as this data is not stored forever.

Reply:
ack